src/api/PDFDocument.ts

```diff
@@ -758,14 +758,19 @@ export default class PDFDocument {
     await srcDoc.flush();
     const copier = PDFObjectCopier.for(srcDoc.context, this.context);
     const srcPages = srcDoc.getPages();
-    const copiedPages: PDFPage[] = new Array(indices.length);
-    for (let idx = 0, len = indices.length; idx < len; idx++) {
-      const srcPage = srcPages[indices[idx]];
-      const copiedPage = copier.copy(srcPage.node);
-      const ref = this.context.register(copiedPage);
-      copiedPages[idx] = PDFPage.of(copiedPage, ref, this);
+    const copiedPagesPromises: Promise<PDFPage>[] = [];
+
+    for (const idx of indices) {
```

Comment:
This is quite ugly. A better way would be:

```
const copiedPages = indixes
  .map(i => srcPages[i])
  .map(page => copier.copy(page.node))
  .map(copy => PDFPage.of(copy, this.context.register(copiedPage), this))
return Promise.all(copiedPages)
```

Reply:
```js
const copiedPages = indices
  .map(i => srcPages[i])
  .map(page => copier.copy(page.node))
  .map(copy => PDFPage.of(copy, this.context.register(copy), this))
return copiedPages
```
the Promise.all isn't necessary

Reply:
```js
const copiedPages = indices
  .map(i => srcPages[i])
  .map(async page => copier.copy(page.node))
  .map(p => p.then(copy => PDFPage.of(copy, this.context.register(copy), this)))
return Promise.all(copiedPages)
```

Reply:
I attempted to use your code, and indices was mispelled which I fixed, but I think this.context.register is supposed to take page as an argument but it's not available in that scope?
```
    const copiedPages = indices
    .map(i => srcPages[i])
    .map(page => copier.copy(page.node))
    .map(copy => PDFPage.of(copy, this.context.register(page), this))
```

Reply:
@MatheusrdSantos didn't see your comment until after I posted mine. I amended to your version. TY!

Reply:
Did a manual test by modifying the code I work on (https://github.com/momijizukamori) manually in node modules, and I didn't see any issues with out of order pages which would be my biggest concern. 

We are hoping to switch over from the original pdf-lib to your version for our project, and if pdf-merging were a little faster that would be a strong motivation to do the switch.